testng-core/src/main/java/org/testng/TestRunner.java

```diff
@@ -931,10 +932,15 @@ private void logStart() {
    *     finish
    */
   private void fireEvent(boolean isStart) {
-    for (ITestListener itl : m_testListeners) {
-      if (isStart) {
+    if (isStart) {
+      for (ITestListener itl : m_testListeners) {
         itl.onStart(this);
-      } else {
+      }
+
+    } else {
+      List<ITestListener> m_testListeners_reverted = new ArrayList<>(m_testListeners);
```

Comment:
This is just a local variable. We can stick to just `testListenersReversed` instead of using the `_` notation.
Historically TestNG has used `m_` to distinguish between data members and local variables (although new code doesnt adhere to this practice)